src/geo/Geography.jsx

```diff
@@ -147,18 +182,17 @@ class Geography extends Component {
     return this.setStateAndRoute({
       items,
       details: {},
-      isLargeFile: this.checkIfLargeFile(this.state.category, items)
     })
   }
 
   onInstitutionChange(selectedLEIs = []){
     let leis = selectedLEIs.map(l => l.value)
     if(leis.includes('all')) leis = []
 
-    return this.setStateAndRoute({
-      leis,
-      details: {},
-      isLargeFile: this.checkIfLargeFile(this.state.category, leis)
+    return this.setState({leis, details: {}}, () => {
+      return this.setStateAndRoute({ 
```

Comment:
Why was this two-fold setState necessary?

Reply:
This was needed because of the way we now calculate the dataset size warning.  

Anytime there are LEIs selected, the record counts from the fetched Institution data take precedence over the pre-existing Geography based counts. When making the transition from 'having selected LEIs' to 'not having selected LEIs' we want to defer calculation to the selected Geography but need to have the previously selected LEIs cleared from the app state before the calculation gets triggered to make it all work.  ⚙️ 

https://github.com/cfpb/hmda-data-browser/blob/filter-leis-by-geography/src/geo/Geography.jsx#L145-L154